pubsub/gochannel/pubsub.go

```diff
@@ -26,6 +30,10 @@ type Config struct {
 	// When true, Publish will block until subscriber Ack's the message.
 	// If there are no subscribers, Publish will not block (also when Persistent is true).
 	BlockPublishUntilSubscriberAck bool
+
+	// When true, messages sent to a topic without any subscribers will be sent to the
+	// subscribers of the `*` topic.
+	EnableFallback bool
```

Comment:
also, food for thought, but I wonder if this fallback couldn't be a closure 🤔

like `NoSubscribersFallback func(msg message.Message, pubsub *GoChannel) error `

It could be much more flexible 🤔 WDYT?

Reply:
That sounds worthwhile to investigate.

Reply:
I investigated this with a private project of mine where I'm using the fork and the closure is awkward to use because the function you want to call has to be known and available when instantiating the `GoChannel`. However, I'm passing this `GoChannel` object around for subscribing and publishing and then just use

```go
noSubscribersMessages, err := pubSub.Subscribe(ctx, gochannel.NoSubscribersFallbackDefaultTopic)
if err != nil {
	return fmt.Errorf("subscribing to fallback topic: %w", err)
}
go func(ctx context.Context, messages <-chan *message.Message) {
	for {
		select {
		case msg := <-messages:
			// handle message

			// lastly:
			msg.Ack()
		case <-ctx.Done():
			return
		}
	}
}(ctx, noSubscribersMessages)
```

which works well and is easy to use. That is to say, I'd leave this as is.